### PR TITLE
[DEV-3751] openapi schema rename for set for set pull requests

### PR DESCRIPTION
**Pull request by gaetano-guerriero**

### CODE REVIEW

File: docs/generate_server.sh [hunk 1]
```diff
@@ -1,3 +1,3 @@
 #!/bin/sh
-docker run --rm -v ${PWD}:/local openapitools/openapi-generator-cli generate -i /local/spec/openapi.yaml -g python-aiohttp -o /local/server_new --git-repo-id athenianco/athenian-api --package-name athenian.api
+docker run --rm -v ${PWD}:/local openapitools/openapi-generator-cli generate -i /local/docs/openapi.yaml -g python-aiohttp -o /local/server_new --git-repo-id athenianco/athenian-api --package-name athenian.api
```
Comment (vmarkovtsev) [inline docs/generate_server.sh:2]: Right, this was from the times when `/spec/openapi.yaml` was a symbolic link to `/server/athenian/api/openapi/openapi.yaml`

File: server/athenian/api/controllers/metrics_controller.py [hunk 2]
```diff
@@ -31,16 +31,16 @@
...
+    ForSetDevelopers, ForSetPullRequests, PullRequestMetricID, ReleaseMetricsRequest
 from athenian.api.models.web.invalid_request_error import InvalidRequestError
 from athenian.api.models.web.pull_request_metrics_request import PullRequestMetricsRequest
 from athenian.api.request import AthenianWebRequest
 from athenian.api.response import model_response, ResponseError
 from athenian.api.tracing import sentry_span
 
 #               service                          developers                                    originals  # noqa
-FilterPRs = Tuple[str, Tuple[List[Set[str]], List[PRParticipants], LabelFilter, JIRAFilter, int, ForSet]]  # noqa
+FilterPRs = Tuple[str, Tuple[List[Set[str]], List[PRParticipants], LabelFilter, JIRAFilter, int, ForSetPullRequests]]  # noqa
```
Comment (vmarkovtsev) [inline server/athenian/api/controllers/metrics_controller.py:43]: These lines went totally out of control. Let's refactor the types in the future DEV-3968

File: server/athenian/api/models/web/calculated_pull_request_metrics_item.py [hunk 3]
```diff
@@ -37,7 +37,7 @@ def for_(self):
         """Gets the for_ of this CalculatedPullRequestMetricsItem.
 
         :return: The for of this CalculatedPullRequestMetricsItem.
-        :rtype: ForSet
+        :rtype: ForSetPullRequests
```
Comment (vmarkovtsev) [inline server/athenian/api/models/web/calculated_pull_request_metrics_item.py:40]: Actually, this should go to a proper type hint in the function signature. Was left from 2 years ago.

File: server/athenian/api/models/web/for_set_developers.py [hunk 4]
```diff
@@ -1,7 +1,8 @@
 from typing import List, Optional
 
 from athenian.api.models.web.base_model_ import AllOf, Model
-from athenian.api.models.web.for_set import CommonPullRequestFilters, RepositoryGroupsMixin
+from athenian.api.models.web.for_set_pull_requests import \
+    CommonPullRequestFilters, RepositoryGroupsMixin
```
Comment (vmarkovtsev) [inline server/athenian/api/models/web/for_set_developers.py:5]: Let's put `CommonPullRequestFilters` and others to `for_set_common.py` or equivalent.

**Review by vmarkovtsev — CHANGES_REQUESTED** [review]
:muscle:

File: server/athenian/api/models/web/calculated_pull_request_histogram.py [hunk 5]
```diff
@@ -55,15 +55,15 @@ def __init__(
         self._interquartile = interquartile
 
     @property
-    def for_(self) -> ForSet:
+    def for_(self) -> ForSetPullRequests:
```
Comment (gaetano-guerriero) [inline server/athenian/api/models/web/calculated_pull_request_histogram.py:58]: I considered this right and typing in `__init__` wrong according to discussion here
https://github.com/athenianco/athenian-api/blob/master/server/athenian/api/controllers/metrics_controller.py#L87

Comment (gaetano-guerriero) [reply server/athenian/api/models/web/calculated_pull_request_histogram.py:58]: `ForSetPullRequests` is a type variable so it's not valid for mypy (like `ForSetCodeChecks`) but this is not obvious to fix

**Review by vmarkovtsev — APPROVED** [review]

**Pull request merged**